[0043]  daemon@ATHENA.MIT.EDU (Peter MacDonald) Linux_Activists 11/11/91 14:06 (35 lines)
Subject: shoelace patch#3: put environment ptr where linux can find it.
Date: Mon, 11 Nov 91 10:58:22 PST
From: pmacdona@sol.UVic.CA (Peter MacDonald)
To: linux-activists@joker.cs.hut.fi

I know, I know.  Three patches in 24 hours... I plead end of term 
instability.   The following one line patch puts the environment
offset relative to the shoelace corpse which is located at segment 0x80000.
Thus, an offset of 0x1234 would put the address of the env at 0x81234.
This offset is stored at location 0x90506: ie. in the word before the 
root device word (508) in the bootsect.s corpse.

By including the shoelace header file shoeconf.h, the linux kernel
could now read in any of the environment variables it wanted (including
scr_cols and scr_rows ;-)

BTW:  Given Bruce Evans comments about shoelace:fsck ties to minix,
we may wish to remove it from the code.  Who wants a facility that
only works with ~20Meg or less partitions anyways :-)


pmacdona@sol.uvic.ca 

--PULL_TAB_HERE -------------------------------
*** shoe.c.bad2	Thu Nov  7 09:33:34 1991
--- shoe.c	Thu Nov  7 09:45:28 1991
***************
*** 650,655 ****
--- 650,656 ----
            else
              *myptr = 0x21c;
        }
+       *--myptr=(unsigned int)SaveConfigPtr;
        Linux = 1;
        LoadPoint = 0x90000;
        bpinx = 0;
--[0043]--
